pilot/pkg/serviceregistry/kube/conversion.go

```diff
@@ -46,12 +46,24 @@ const (
 	managementPortPrefix = "mgmt-"
 )
 
-func convertPort(port coreV1.ServicePort) *model.Port {
-	return &model.Port{
-		Name:     port.Name,
-		Port:     int(port.Port),
-		Protocol: kube.ConvertProtocol(port.Port, port.Name, port.Protocol),
+func convertPort(port coreV1.ServicePort) []*model.Port {
+	out := []*model.Port{
+		{
+			Name:     port.Name,
+			Port:     int(port.Port),
+			Protocol: kube.ConvertProtocol(port.Port, port.Name, port.Protocol),
+		},
+	}
+
+	if port.NodePort > 0 {
+		out = append(out, &model.Port{
+			Name:     port.Name,
+			Port:     int(port.NodePort),
```

Comment:
should we augment port to have nodePort as a field rather than creating a new port?

Reply:
Beacsue we expect each port to have a different name

Reply:
Two reasons why i did not do that:

1. the nodeport is only for k8s platform, but model.Port is an abstract for all.

2.  Actually the 80 and 31380 is a same port.
 ```
  - name: http2
    nodePort: 31380
    port: 80
    protocol: TCP
    targetPort: 80
```